code/game/machinery/antibody.dm

```diff
@@ -0,0 +1,134 @@
+#define MAX_STAGES 3
+
+/obj/machinery/antibody_extractor
+	name = "antibody extractor"
+	desc = "A machine made to extract antibodies. There's a holographic info-label on it..."
+	desc_extended = "This machine is made to extract antibodies from a subject. It will only work with the <b>unmutated, original</b> version of the virus. \
+					<span class='danger'>This machine will immediately notify the entire facility if used without authorization.</span>"
+	icon = 'icons/obj/machinery/extractor.dmi'
+	icon_state = "extractor"
+	anchored = TRUE
+	can_buckle = list(/mob/living/carbon/human)
+	/// If the machine is currently under use.
+	var/working = FALSE
+	/// Current stage.
+	var/stage = 0
+	/// Current progress to the next stage.
+	var/cure_progress = 0
+	/// Threshold to the next stage.
+	var/stage_threshold = 100
+	/// The person from which the cure is being extracted.
+	var/mob/living/carbon/human/occupant
+
+/obj/machinery/antibody_extractor/Destroy()
+	occupant = null
+	return ..()
+
+/obj/machinery/antibody_extractor/attack_hand(mob/user)
+	if(working)
+		to_chat(user, SPAN_WARNING("\The [src] is totally unresponsive while processing the cure!"))
+		return
+
+	if(user == occupant)
+		to_chat(user, SPAN_WARNING("The buckles stop you from doing much at all."))
+		. = ..()
```

Comment:
you're assigning the parent return value to . here, and then returning FALSE anyway. ..() would work here, or you could just remove this line depending on what attack_hand should do